### PR TITLE
Tests for editTimetableMetadata endpoint

### PR DESCRIPTION
**Pull request by TheComputerM**

### CODE REVIEW

**Comment by TheComputerM** [comment]
Some questions:
- Is it really necessary to hard-code the response body messages in the tests
- Can we call the request directly in the `describe` block instead of calling it in a `test` block  and doing `response: Response | null`. Even if we do it in a seperate block, why not use `beforeAll` to call the api endpoint?

PS: thanks karthik I copied a ton of your stuff

**Comment by majimearun** [comment]
> Some questions:
> 
>     * Is it really necessary to hard-code the response body messages in the tests
> 
>     * Can we call the request directly in the `describe` block instead of calling it in a `test` block  and doing `response: Response | null`. Even if we do it in a seperate block, why not use `beforeAll` to call the api endpoint?
> 
> 
> PS: thanks karthik I copied a ton of your stuff

Lmao okay first off, yes thank you karthik, I changed my testing code structure now too, cz his looked neat af.

For your questions

1. what is your alternative for hardcoding the response error messages? A lookup from some json?
2. From what I've seen, nothing except variable declaration is done directly in the describe block, if that is what you're suggesting.

File: src/controllers/timetable/editTimetableMetadata.test.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            timetables: [],
+          }))
+        )
+        .execute();
+    });
+
+    it("Store test user uuids", async () => {
+      const users = await userRepository
+        .createQueryBuilder("user")
+        .orderBy("user.batch")
```
Comment (majimearun) [inline src/controllers/timetable/editTimetableMetadata.test.ts:57]: Does it need to be ordered

File: src/controllers/timetable/editTimetableMetadata.test.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+          .createQueryBuilder()
+          .insert()
+          .into(Timetable)
+          .values([
+            {
+              name: "Draft Timetable",
+              authorId: uuids[i],
+              degrees: mockUserData[i].degrees,
+              private: false,
+              draft: true,
```
Comment (majimearun) [inline src/controllers/timetable/editTimetableMetadata.test.ts:74]: This state is invalid according to our flow, a timetable can only be private if it is a draft

File: src/controllers/timetable/editTimetableMetadata.test.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+              warnings: [],
+              createdAt: new Date(),
+              lastUpdated: new Date(),
+            },
+            {
+              name: "Final Timetable",
+              authorId: uuids[i],
+              degrees: mockUserData[i].degrees,
+              private: true,
+              draft: false,
```
Comment (majimearun) [inline src/controllers/timetable/editTimetableMetadata.test.ts:91]: Same timetable state problem

File: src/controllers/timetable/editTimetableMetadata.test.ts [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+            },
+          ])
+          .execute();
+      }
+    });
+
+    it("Store test timetables", async () => {
+      userData = await userRepository
+        .createQueryBuilder("user")
+        .orderBy("user.batch")
```
Comment (majimearun) [inline src/controllers/timetable/editTimetableMetadata.test.ts:111]: Again, is the order by required

File: src/controllers/timetable/editTimetableMetadata.ts [hunk 5]
```diff
@@ -28,45 +28,45 @@ const dataSchema = z.object({
...
-    return res.status(401).json({ message: "unregistered user" });
+  if (!timetable) {
+    return res.status(404).json({ message: "timetable not found" });
   }
 
-  const id: number = parseInt(req.params.id);
+  let author: User | null = null;
 
-  let timetable: Timetable | null = null;
+  const name: string = req.body.name;
```
Comment (majimearun) [inline src/controllers/timetable/editTimetableMetadata.ts:53]: Is there a reason you rearranged these lmao

**Review by majimearun — CHANGES_REQUESTED** [review]
Asked for a couple changes, and there will be some more tests you'll need to write after the edit flow (#47) is merged.

File: src/controllers/timetable/editTimetableMetadata.ts (hunk 5 above)
Comment (TheComputerM) [reply src/controllers/timetable/editTimetableMetadata.ts:53]: Logically, I think it should check if the timetable exists before it checks if the user is the owner of the timetable etc.

File: src/controllers/timetable/editTimetableMetadata.test.ts (hunk 4 above)
Comment (TheComputerM) [reply src/controllers/timetable/editTimetableMetadata.test.ts:111]: Yeah, to test if I have updated the timetable, I store the old state and get the new state and compare them. It's would be error free if the states are in the same order.

File: src/controllers/timetable/editTimetableMetadata.test.ts (hunk 2 above)
Comment (TheComputerM) [reply src/controllers/timetable/editTimetableMetadata.test.ts:74]: Hmm, will update

File: src/controllers/timetable/editTimetableMetadata.test.ts (hunk 1 above)
Comment (TheComputerM) [reply src/controllers/timetable/editTimetableMetadata.test.ts:57]: See my other comment: https://github.com/crux-bphc/chronofactorem-rewrite/pull/50#discussion_r1278557568

File: src/controllers/timetable/editTimetableMetadata.test.ts (hunk 4 above)
Comment (majimearun) [reply src/controllers/timetable/editTimetableMetadata.test.ts:111]: yeapp okay makes sense

**Comment by majimearun** [comment]
PR #47 has been merged

**Comment by soumitradev** [comment]
Closing because stale, and too many changes to backend code since this PR

**Pull request closed without merging**